src/services/NoteService.ts

```diff
@@ -0,0 +1,75 @@
+import { User } from '../entities/User';
+import { getConnection, getRepository } from 'typeorm';
+import { singleton } from 'tsyringe';
+import { APIError, formatValidationErrors, HttpCode } from '../util/errors';
+import { validateOrReject } from 'class-validator';
+import Note, { APINote } from '../entities/Note';
+
+export type NoteDataToCreate = Pick<Note, 'noteType' >;
+
+enum NoteUserError {
+	NoteTypeNotFound = 'Note Type not found',
+	UserNotFound = 'User not found',
+	InvalidEntryDetails = 'Invalid user details.',
+	NoteBadRequest = 'Bad request'
+}
+
+@singleton()
+export class NoteService {
+	public async getNotes(userID: string): Promise<APINote[]> {
+		const notes = await getRepository(Note)
+			.createQueryBuilder('note')
+			.leftJoinAndSelect('note.owner', 'user')
+			.where('user.id = :id', { id: userID })
+			.leftJoinAndSelect('note.targetUser', 'targetUser')
+			.getMany();
+		return notes.map(note => note.toJSON());
+	}
+
+	public async createNote(userID: string, targetUserID: string, data: NoteDataToCreate): Promise<APINote> {
+		return getConnection().transaction(async entityManager => {
+			if (!userID) throw new APIError(HttpCode.NotFound, NoteUserError.UserNotFound);
+			if (!targetUserID) throw new APIError(HttpCode.NotFound, NoteUserError.UserNotFound);
+			const owner = await entityManager.findOneOrFail(User, { id: userID })
+				.catch(() => Promise.reject(new APIError(HttpCode.NotFound, NoteUserError.UserNotFound)));
+			const targetUser = await entityManager.findOneOrFail(User, { id: targetUserID })
+				.catch(() => Promise.reject(new APIError(HttpCode.NotFound, NoteUserError.UserNotFound)));
+
+			let note: Note | undefined;
+			note = await getRepository(Note)
+				.createQueryBuilder('note')
+				.leftJoinAndSelect('note.owner', 'user')
+				.where('user.id = :id', { id: userID })
+				.leftJoinAndSelect('note.targetUser', 'targetUser')
```

Comment:
An inner join would work better here [https://github.com/typeorm/typeorm/blob/master/docs/select-query-builder.md#inner-and-left-joins](https://github.com/typeorm/typeorm/blob/master/docs/select-query-builder.md#inner-and-left-joins). It allows us to join if a criteria is met.